src/sharedData/visualization/components/Visualization.js

```diff
@@ -352,6 +303,48 @@ const MapboxLayer = props => {
     visualizationConfig?.annotateConfig?.annotationTitle,
   ]);
 
+  useEffect(() => {
+    if (!map) {
+      return;
+    }
+    const configBounds = (function () {
+      const viewportBounds = visualizationConfig?.viewportConfig?.bounds;
+      if (!viewportBounds) {
+        return;
+      }
+      const southWest = viewportBounds.southWest;
+      const northEast = viewportBounds.northEast;
+      if (!southWest || !northEast) {
+        return;
+      }
+      const sw = new mapboxgl.LngLat(southWest.lng, southWest.lat);
+      const ne = new mapboxgl.LngLat(northEast.lng, northEast.lat);
+      return new mapboxgl.LngLatBounds(sw, ne);
+    })();
+
+    const geoJsonBounds = (function () {
+      const source = map.getSource('visualization');
+      if (!source || !source._data) {
+        return;
+      }
+      const calculatedBbox = bbox(map.getSource('visualization')._data);
+      return new mapboxgl.LngLatBounds(
+        [calculatedBbox[0], calculatedBbox[1]],
+        [calculatedBbox[2], calculatedBbox[3]]
+      );
+    })();
+
+    const bounds = useConfigBounds ? configBounds : geoJsonBounds;
```

Comment:
question: what does this do? when do you use config bounds vs JSON bounds?

Reply:
@paulschreiber we have 2 type of bounds:
1. GeoJSON bounds: These are the bounds calculated from the geo json data
2. Config bounds: These are the bounds explicitly set by the user when they move the viewport

For some steps we don't need to use the config boungs, just the geojson bounds like the annotate step where just one point is displayed

Reply:
Should we call it viewportBounds instead of configBounds?

Reply:
@paulschreiber for me it makes more sense configBounds, because ideally with our model the bounds could be defined in different ways, and the idea for this condition is to use or not use the bounds defined in the visualization configuration, I think how the user defined those bounds is relevant here

Reply:
Maybe visualizationConfigBounds then? ConfigBounds sounded like it was based on the application configuration.